reclass/datatypes/tests/test_applications.py

```diff
@@ -63,7 +63,7 @@ def test_repr_negations_interspersed(self):
         l = ['a', '~b', 'a', '~d']
         a = Applications(l)
         is_negation = lambda x: x.startswith(a.negation_prefix)
-        GOAL = filter(lambda x: not is_negation(x), set(l)) + filter(is_negation, l)
+        GOAL = list(filter(lambda x: not is_negation(x), set(l))) + list(filter(is_negation, l))
```

Comment:
Is that really worth it for what is just test code, operating on very small lists? You'll have to be more specific with your other suggestions, it's not immediately clear to me how any functions in those modules would help out here. I could also replace it with list comprehensions.